MapboxCoreNavigation/SpokenInstructionFormatter.swift

```diff
@@ -76,7 +76,12 @@ public class SpokenInstructionFormatter: NSObject {
                 text = String.localizedStringWithFormat(NSLocalizedString("CONTINUE", bundle: .mapboxCoreNavigation, value: "Continue for %@", comment: "Format for speech string after completing a maneuver and starting a new step; 1 = distance"), escapeIfNecessary(maneuverVoiceDistanceFormatter.string(from: userDistance)))
             }
         } else if alertLevel == .high && upcomingStepDuration < linkedInstructionMultiplier {
-            text = String.localizedStringWithFormat(NSLocalizedString("LINKED_UTTERANCE_FORMAT", bundle: .mapboxCoreNavigation, value: "%@, then %@", comment: "Format for speech string; 1 = current instruction; 2 = the following linked instruction"), upComingInstruction, followOnInstruction)
+            // If the user is entering the roundabout and there is exit information, don't link instruction
+            if let upcomingStep = routeProgress.currentLegProgress.upComingStep, (upcomingStep.maneuverType == .takeRotary || upcomingStep.maneuverType == .takeRoundabout) && upcomingStep.exitIndex != nil {
```

Comment:
Can we simplify this logic by waiting until Project-OSRM/osrm-backend#4358 is deployed and removing this special case altogether?

Reply:
@1ec5 yep, that's the idea. This PR would be a regression until then. We'll also need to update osrm-text-instructions to account for the new type.

Reply:
I think this code would read more naturally as:

```swift
if followOnInstruction.maneuverType == .exitRoundabout || followOnInstruction.maneuverType == .exitRotary`
```

That way, if the developer fails to specify `includesRoundaboutExitManeuver` in the original route options and there’s a turn close to the exit, that turn won’t be omitted entirely.